### PR TITLE
feat(discovery): provide an option to disable discovery debug log

### LINKED ISSUE #1935: 求助，怎么能关闭这个 info 日志？
https://github.com/go-kratos/kratos/blob/d0b704b8f38db901488092d0982c114574da76a7/transport/grpc/resolver/discovery/resolver.go#L81

收集的日志一堆的这种日志：
INFO msg=[resolver] update instances: 

能提供配置来关闭这种日志的打印吗？

### PR DESCRIPTION
**Pull request by yeqown**
#### Description (what this PR does / why we need it):

Issue resolved

#### Which issue(s) this PR fixes (resolves / be part of):

fixes #1935

### CODE REVIEW

**Comment by haiyux** [comment]
What's this for？ Do not output resolver log？

**Comment by yeqown** [comment]
@haiyux  provide an option to disable "[resolve] update instance xxx" log which is abundant and intrusive for someone and it's unstoppable now.

**Comment by Casper-Mars** [comment]
How about passing a no-output logger ? I think this is more elegant.

File: transport/grpc/resolver/discovery/builder.go [hunk 1]
```diff
@@ -37,20 +38,29 @@ func WithInsecure(insecure bool) Option {
 	}
 }
 
+// WithDebugLogDisabled 屏蔽调试日志
+func WithDebugLogDisabled() Option {
```
Comment (tonybase) [inline transport/grpc/resolver/discovery/builder.go:42]: DisableDebugLog()

**Comment by yeqown** [comment]
> How about passing a no-output logger ? I think this is more elegant.

you can‘t even know whether the discoveryResolver started or not.

File: transport/grpc/resolver/discovery/builder.go (hunk 1 above)
Comment (yeqown) [reply transport/grpc/resolver/discovery/builder.go]: updated

**Review by tonybase — APPROVED** [review]

**Comment by freezeChen** [comment]
新提供的方法DisableDebugLog()隐藏在grpc client DialInsecure()方法的内部,外部并没办法对discovery.NewBuilder()进行参数设置,  建议在 grpc client DialInsecure()的options中添加 debuglog的开关,传递至discovery.NewBuilder内部

**Pull request merged**